clients/clients-integration-tests/src/test/java/org/apache/kafka/clients/producer/ProducerSendWhileDeletionTest.java

```
@@ -157,6 +164,42 @@ public void testSendWithRecreatedTopic() throws Exception {
         }
     }
 
+    @Timeout(90)
+    @ClusterTest
+    public void testSendWhileTopicGetRecreated() {
+        int maxNumRecreatTopicAttempts = 10;
+        List<Uuid> topicIds = new CopyOnWriteArrayList<>();
+        var recreateTopicFuture = CompletableFuture.runAsync(() -> {
+            for (int i = 1; i <= maxNumRecreatTopicAttempts; i++) {
+                Uuid topicId = recreateTopic();
+                if (topicId != Uuid.ZERO_UUID) {
+                    topicIds.add(topicId);
+                }
+            }
+        });
+
+        AtomicInteger numSuccess = new AtomicInteger(0);
+        var producerFuture = CompletableFuture.runAsync(() -> {
+            try (var producer = cluster.producer()) {
+                for (int i = 1; i <= numRecords; i++) {
+                    var resp = producer.send(new ProducerRecord<>(topic, null, ("value" + i).getBytes()),
+                            (metadata, exception) -> {
+                                if (metadata != null) {
+                                    numSuccess.incrementAndGet();
+                                }
+                            }).get();
+                    assertEquals(resp.topic(), topic);
+                }
+            } catch (Exception e) {
+                // ignore
```

Are we sure it's safe to ignore errors here? What errors are we expecting to see that are safely ignorable?

---

We don't care much as long as we are recreating enough topic ids for the same topic name at the end. This code has been updated with better retry mechanisme now